operatorcourier/validate.py

```diff
@@ -167,20 +172,54 @@ def _csv_spec_validation(self, spec, bundleData):
                     pass
 
             if "owned" not in customresourcedefinitions:
-                self._log_error("spec.customresourcedefinitions.owned"
+                self._log_error("spec.customresourcedefinitions.owned "
                                 "not defined for csv")
-                valid = False
-            else:
-                for crd in customresourcedefinitions["owned"]:
-                    if "name" not in crd:
-                        self._log_error("name not defined for item in "
-                                        "spec.customresourcedefinitions.")
-                        valid = False
-                    else:
-                        if crd["name"] not in crdList:
-                            self._log_error("custom resource definition referenced "
-                                            "in csv not defined in root list of crds")
+                return False
+
+            for csvOwnedCrd in customresourcedefinitions["owned"]:
+                if "name" not in csvOwnedCrd:
+                    self._log_error("name not defined for item in "
+                                    "spec.customresourcedefinitions.")
+                    valid = False
+                elif csvOwnedCrd["name"] not in crdList:
+                    self._log_error("custom resource definition referenced in csv "
+                                    "not defined in root list of crds")
+                    valid = False
 
+                if "kind" not in csvOwnedCrd:
+                    self._log_error("kind not defined for item in "
+                                    "spec.customresourcedefinitions.")
+                    valid = False
+                if "version" not in csvOwnedCrd:
+                    self._log_error("version not defined for item in "
+                                    "spec.customresourcedefinitions.")
+                    valid = False
+
+                for crd in bundleData[self.crdKey]:
+                    if 'name' not in csvOwnedCrd or not \
```

Comment:
nit: This if statement is kinda long, and I'm not sure what it's checking for: Are we continuing if the name doesn't exist or if it does exist?

Reply:
We continue to check field matches only if the name in both CRD and CSV exists, and they match.